### PR TITLE
Refactor investigator info

### PR DESCRIPTION
**Pull request by ruaraya**
Refactor relationships defined for private investigator. The related tests will be adjusted according to ticket: https://jag.gov.bc.ca/jira/browse/SPDBT-2796

### CODE REVIEW

File: src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs [hunk 1]
```diff
@@ -162,43 +166,62 @@ private void LinkOrganization(Guid? accountId, spd_application app)
...
+
+        var position = _context.LookupPosition(PositionEnum.PrivateInvestigatorManager.ToString());
+
+        if (position != null)
+            _context.AddLink(position, nameof(spd_businesscontact.spd_position_spd_businesscontact), bizContact);
+    }
+
+    private void DeletePrivateInvestigatorLink(spd_application app)
+    {
+        spd_businesscontact? bizContact = app.spd_businesscontact_spd_application.FirstOrDefault();
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs:211]: one application can have many bizContact, how do you know you delete the priviateInvestigator one?

Comment (ruaraya) [reply src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs]: Nice catch, I adjusted the query to make sure the record that has the private investigator is pulled

File: src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs [hunk 2]
```diff
@@ -162,43 +166,66 @@ private void LinkOrganization(Guid? accountId, spd_application app)
         }
     }
 
-    private void LinkPrivateInvestigator(SwlContactInfo privateInvestigatorInfo, spd_application app)
+    private spd_businesscontact UpsertPrivateInvestigator(PrivateInvestigatorSwlContactInfo privateInvestigatorInfo, spd_application app)
     {
-        if (privateInvestigatorInfo.LicenceId == null)
-            return;
+        spd_businesscontact? bizContact = null;
+        DeletePrivateInvestigatorLink(app);
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs:172]: why we need to do this?

File: src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs [hunk 3]
```diff
@@ -58,13 +59,13 @@ public async Task<BizLicApplicationCmdResp> CreateBizLicApplicationAsync(CreateB
...
-        //else
-        //    DeletePrivateInvestigatorLink(cmd.PrivateInvestigatorSwlInfo, app);
-
-        await _context.SaveChangesAsync(ct);
+        if (cmd.CategoryCodes.Any(c => c == WorkerCategoryTypeEnum.PrivateInvestigator))
+        {
+            contact contact = GetContact((Guid)cmd.PrivateInvestigatorSwlInfo.ContactId);
+            spd_businesscontact businessContact = UpsertPrivateInvestigator(cmd.PrivateInvestigatorSwlInfo, app);
+            _context.SetLink(businessContact, nameof(spd_businesscontact.spd_ContactId), contact);
+            await _context.SaveChangesAsync(ct);
```
[the remark this answers is not in the document]
Comment (ruaraya) [reply src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs]: Fixed

File: src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs (hunk 2 above)
Comment (ruaraya) [reply src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs:171]: The idea is to remove all relationships (links) around business contact before adding or updating a new business contact

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**